core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -213,18 +217,22 @@ public Set<TrustAnchor> getTrustAnchors() {
                 .stream()
                 .filter(obj -> obj instanceof JsonObject)
                 .map(JsonObject.class::cast)
-                .map(this::getTrustAnchorForPublicKey)
+                .map(this::getTrustAnchorForTrustedCA)
                 .filter(anchor -> anchor != null)
                 .collect(Collectors.toSet());
         }
         return trustAnchors;
     }
 
     @JsonIgnore
-    private TrustAnchor getTrustAnchorForPublicKey(final JsonObject keyProps) {
+    private TrustAnchor getTrustAnchorForTrustedCA(final JsonObject keyProps) {
```

Comment:
can we please keep creating the trust anchor from the CN and public key? We deliberately chose to not persist the encoded certificate but the PK and CN only ...

Reply:
[OpenJDK implementation](https://github.com/openjdk/jdk/blob/jdk-17%2B35/src/java.base/share/classes/sun/security/provider/certpath/OCSPResponse.java#L417) requires full X509 certificate for OCSP response verification, in most cases it will match the issuer CA, so the easiest way was to persist original certificate. However, we could always explicitly set [ocspResponderCert](https://github.com/eclipse-hono/hono/blob/73472a4ee8a7c235cbee998f67a4f61b06cbaf70/services/device-registry-base/src/main/java/org/eclipse/hono/service/management/tenant/TrustedCertificateAuthority.java#L94) field instead and keep the current current trust anchor setup. Should I rather change it in this way?